packages/onboarding/src/AssertUtil.ts

```diff
@@ -0,0 +1,488 @@
+  iotInfoKey,
+  mobileInfoKey,
+  rewardableEntityConfigKey,
+} from '@helium/helium-entity-manager-sdk'
+import * as Currency from '@helium/currency-utils'
+import {
+  BONES_IN_HNT,
+  DcProgram,
+  FULL_LOCATION_STAKING_FEE,
+  HemProgram,
+  HotspotType,
+  TXN_FEE_IN_LAMPORTS,
+} from './types'
+
+type Account = AccountInfo<string[]>
+
+const getOraclePriceFromSolana = async (opts: { connection: Connection; cluster: Cluster }) => {
+  const price = await Currency.getOraclePrice({ tokenType: 'HNT', ...opts })
+  if (!price?.aggregate.price) {
+    throw new Error('Failed to fetch oracle price')
+  }
+
+  return new BN(price.aggregate.price)
+}
+
+const getBalance = async (wallet: PublicKey, connection: Connection, mint: PublicKey) => {
+  const balance = await Currency.getBalance({
+    pubKey: wallet,
+    connection,
+    mint: new PublicKey(mint),
+  })
+  return new BN(balance.toString())
+}
+
+const getBalances = async (wallet: PublicKey, connection: Connection) => {
+  const sol = await connection.getBalance(wallet)
+  const hnt = (await getBalance(wallet, connection, HNT_MINT)).div(BONES_IN_HNT)
+  const dc = await getBalance(wallet, connection, DC_MINT)
+
+  return { hnt, dc, sol: new BN(sol) }
+}
+
+const burnHNTForDataCredits = async ({
+  dcAmount,
+  wallet,
+  connection,
+  dcProgram,
+}: {
+  dcAmount: BN
+  wallet: PublicKey
+  connection: Connection
+  dcProgram: DcProgram
+}) => {
+  const txn = await dcProgram.methods
+    .mintDataCreditsV0({
+      hntAmount: null,
+      dcAmount,
+    })
+    .preInstructions([
+      createAssociatedTokenAccountIdempotentInstruction(
+        wallet,
+        getAssociatedTokenAddressSync(DC_MINT, wallet, true),
+        wallet,
+        DC_MINT,
+      ),
+    ])
+    .accounts({
+      dcMint: DC_MINT,
+      recipient: wallet,
+    })
+    .transaction()
+
+  if (!txn) return
+
+  const { blockhash } = await connection.getLatestBlockhash('finalized')
+  txn.recentBlockhash = blockhash
+  txn.feePayer = wallet
+
+  return txn
+}
+
+const getAtaAccountCreationFee = async (solanaAddress: PublicKey, connection: Connection) => {
+  const ataAddress = getAssociatedTokenAddressSync(DC_MINT, solanaAddress, true)
+
+  try {
+    await getAccount(connection, ataAddress)
+    return new BN(0)
+  } catch {
+    return new BN(2039280)
+  }
+}
+
+const getStakingFeeForType = async (type: HotspotType, hemProgram: HemProgram) => {
+  const isIOT = type === 'IOT'
+  const mint = isIOT ? IOT_MINT : MOBILE_MINT
+  const subDao = subDaoKey(mint)[0]
+
+  const configKey = rewardableEntityConfigKey(subDao, type)
+
+  const entityConfig = await hemProgram?.account.rewardableEntityConfigV0.fetchNullable(
+    configKey[0],
+  )
+  const config = isIOT ? entityConfig?.settings.iotConfig : entityConfig?.settings.mobileConfig
+
+  // @ts-ignore
+  const configFee = config?.fullLocationStakingFee as BN
+  if (configFee) {
+    return toBN(configFee, 0)
+  }
+  return new BN(FULL_LOCATION_STAKING_FEE)
```

Comment:
This will no longer work. https://github.com/helium/wallet-app/blob/main/src/hooks/useOnboardingBalances.ts#L141